cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/PreferredLeaderElectionGoal.java

```diff
@@ -103,50 +103,46 @@ public boolean optimize(ClusterModel clusterModel, Set<Goal> optimizedGoals, Opt
         }
       }
     }
-    // Check whether this goal has relocated any leadership.
-    boolean relocatedLeadership = false;
     Set<Integer> excludedBrokersForLeadership = optimizationOptions.excludedBrokersForLeadership();
-    // Ignore the excluded topics because this goal does not move partitions.
+    // Iterate over partitions and try to move its leader replica to the first replica in replica list.
+    int failedPartitions = 0;
+    StringBuilder sb = new StringBuilder();
     for (List<Partition> partitions : clusterModel.getPartitionsByTopic().values()) {
       for (Partition p : partitions) {
         if (hasBrokerOrDiskToBeDemoted && !partitionsToMove.contains(p.topicPartition())) {
           continue;
         }
-        for (Replica r : p.replicas()) {
-          // Iterate over the replicas and ensure that (1) the leader is set to the first alive replica, and (2) the
-          // leadership is not transferred to a broker excluded for leadership transfer.
-          Broker leaderCandidate = r.broker();
-          if (leaderCandidate.isAlive()) {
-            if (r.isCurrentOffline()) {
-              LOG.warn("The preferred replica of partition {} on broker {} is offline.", p.topicPartition(), leaderCandidate);
-              continue;
-            }
-            if (!r.isLeader()) {
-              if (excludedBrokersForLeadership.contains(leaderCandidate.id())) {
-                LOG.warn("Skipped leadership transfer of partition {} to broker {} because it is among brokers excluded"
-                         + " for leadership {}.", p.topicPartition(), leaderCandidate, excludedBrokersForLeadership);
-                continue;
-              }
-              clusterModel.relocateLeadership(r.topicPartition(), p.leader().broker().id(), leaderCandidate.id());
-              relocatedLeadership = true;
-            }
-            if (clusterModel.demotedBrokers().contains(leaderCandidate)) {
-              LOG.warn("The leader of partition {} has to be on a demoted broker {} because all the alive "
-                       + "replicas are demoted.", p.topicPartition(), leaderCandidate.id());
-            }
-            if (r.disk() != null && r.disk().state() == Disk.State.DEMOTED) {
-              LOG.warn("The leader of partition {} has to be on a demoted disk {} of broker {} because all the alive "
-                       + "replicas are demoted.", p.topicPartition(), r.disk().logDir(), leaderCandidate.id());
-            }
-            break;
+        Replica candidateReplica = p.replicas().get(0);
+        Broker candidateBroker = candidateReplica.broker();
+        if (!candidateReplica.isLeader()) {
```

Comment:
If this goal is called due to a demote request, then (if possible to do so) the leadership should be removed from the current leader regardless of whether it is transferred to the most preferred replica or not.

Overall, rather than having all these changes, can't we just
1. Identify if this request is purely a PLE as opposed to a demote request, and
2. If it is a pure PLE request, skip the partition if its most preferred replica is offline?